cromwell_tools/tests/test_cromwell_api.py

```diff
@@ -0,0 +1,349 @@
+            },
+            "PAPI": {
+                "ok": "true"
+            },
+            "GCS": {
+                "ok": "true"
+            }
+        }
+
+        def _request_callback(request, context):
+            context.status_code = 200
+            context.headers['test'] = 'header'
+            return expected
+
+        for cromwell_auth in self.auth_options:
+            mock_request.get('{0}/api/engine/v1/status'.format(cromwell_auth.url),
+                              json=_request_callback)
+            result = CromwellAPI.health(cromwell_auth)
+            self.assertEqual(result.status_code, 200)
+            self.assertEqual(result.json(), expected)
+
+    @requests_mock.mock()
+    def test_abort(self, mock_request):
+        workflow_id = "01234"
+        expected = {
+            "id": workflow_id,
+            "status": "Aborting"
+        }
+
+        def _request_callback(request, context):
+            context.status_code = 200
+            context.headers['test'] = 'header'
+            return expected
+
+        for cromwell_auth in self.auth_options:
+            mock_request.post(cromwell_auth.url + '/api/workflows/v1/{}/abort'.format(workflow_id),
+                              json=_request_callback)
+            result = CromwellAPI.abort(workflow_id, cromwell_auth)
+            self.assertEqual(result.json(), expected)
+
+    @requests_mock.mock()
+    def test_status(self, mock_request):
+        def _request_callback_status(request, context):
+            context.status_code = 200
+            context.headers['test'] = 'header'
+            return {'status': 'Succeeded'}
+
+        workflow_id = "01234"
+        for cromwell_auth in self.auth_options:
+            mock_request.get(cromwell_auth.url + '/api/workflows/v1/{}/status'.format(workflow_id),
+                             json=_request_callback_status)
+            result = CromwellAPI.status(workflow_id, cromwell_auth)
+            self.assertEqual(result.json()['status'], 'Succeeded')
+
+
+class TestValidate(unittest.TestCase):
+
+    def test_localize_file(self):
+        temporary_directory = tempfile.mkdtemp()
+
+        # test that we can localize both local and https files. Use this file as a convenient target
+        targets = [
+            'https://raw.githubusercontent.com/broadinstitute/cromwell-tools/master/'
+            'cromwell_tools/tests/test_cromwell_tools.py',
+            __file__
+        ]
+        for target in targets:
+            utils._localize_file(target, temporary_directory)
+            localized_file = os.path.join(temporary_directory, os.path.basename(target))
+
+            # verify the file was localized and that it contains some expected content
+            assert os.path.isfile(localized_file)
+            with open(localized_file, 'r') as f:
+                assert 'cromwell_tools' in f.read()
+            os.remove(localized_file)
+
+    # TODO: Fix failing test
+    @unittest.skip
```

Comment:
Will this test pass if you remove this decorator? I think it was just broken when I ran it locally because I did not have `os.environ['WOMTOOL']` set

Reply:
It will pass if you run the tests within the docker image, which we should recommend people to use. In the latest commit, I also make this conditional, so it won't fail anymore if users do not have os.environ['WOMTOOL'] set